app/models/cluster_file_info.rb

```diff
@@ -5,6 +5,22 @@ class ClusterFileInfo
 
   field :custom_colors, type: Hash, default: {}
 
+  # decode base64 annotation/label names back into human-readable text
+  def decoded_custom_colors
+    self.class.transform_custom_colors(custom_colors, :decode64)
+  end
+
+  # helper to transform a nested custom color hash to/from base64
+  def self.transform_custom_colors(hash, transform = :encode64)
```

Comment:
why would we want to call this function with a different transform?  Wouldn't that cause data problems if some clusters were encoded with different transforms?  I think if :encode64 and :decode64 are declared at the top of the file as COLOR_ENCODE_TRANSFORM, it communicates more clearly what we're doing

Reply:
I was initially going to do that, but thought that adding an extra constant just for that seemed overkill.  But your point holds - we only want to allow `:encode64` and `:decode64`, so I'll change that.